recipes/qt/6.x.x/conanfile.py

```diff
@@ -1371,5 +1376,18 @@ def _create_plugin(pluginname, libname, type, requires):
                     obj_files = [os.path.join(submodule_dir, file) for file in os.listdir(submodule_dir)]
                     self.cpp_info.components[component].exelinkflags.extend(obj_files)
                     self.cpp_info.components[component].sharedlinkflags.extend(obj_files)
-
-        self.cpp_info.set_property("cmake_build_modules", build_modules)
+                    
+        build_modules_list = []
+                    
+        def _add_build_modules_for_component(component):
+            for req in self.cpp_info.components[component].requires:
+                if "::" in req: # not a qt component                   
+                    continue
+                _add_build_modules_for_component(req)
+            build_modules_list.append(build_modules.get(component, []))
+            build_modules[component] = []
```

Comment:
```suggestion
        def _add_build_modules_for_component(component):
            nonlocal build_modules_list
            for req in self.cpp_info.components[component].requires:
                if "::" in req: # not a qt component
                    continue
                _add_build_modules_for_component(req)
            build_modules_list += build_modules.get(component, [])
            build_modules[component] = []
```

Reply:
Seems to fix https://github.com/conan-io/conan-center-index/issues/13760 with the suggested changes otherwise you get
> TypeError: join() argument must be str, bytes, or os.PathLike object, not 'list'

And add nonlocal to avoid a UnboundLocalError when using `build_modules_list +=`

Reply:
thanks ! I don't understand how I did not get this error in https://github.com/eirikb/proof-of-conan/actions/runs/3334863393, or in current builds (https://c3i.jfrog.io/artifactory/misc/logs/pr/12906/18-linux-gcc/qt/6.1.3/3affa9a52cf5ffd4a97f16bd2aa7eeede2d38954-build.txt or https://c3i.jfrog.io/artifactory/misc/logs/pr/12906/18-linux-gcc/qt/6.1.3/3affa9a52cf5ffd4a97f16bd2aa7eeede2d38954-test.txt for example).
I fixed the issue by using `extend` https://github.com/conan-io/conan-center-index/pull/12906/commits/739316cc63ea5447dfd29954f8bf927af06fce92 it should be ok right ?

Reply:
Yes your commit also works fine.
I guess ou did not get the error because cmake_build_modules property is only used by CMakeDeps and it does not seem to be used in the test_package. Maybe we could add it to prevent regression.

Reply:
that's a first time I encounter `nonlocal` keyword ever... had to google and read it, never seen it for years